bitnami/kafka/templates/scripts-configmap.yaml

```diff
@@ -125,12 +125,13 @@ data:
     {{- else if eq .Values.externalAccess.service.type "NodePort" }}
     {{- if and .Values.externalAccess.service.usePodIPs .Values.externalAccess.autoDiscovery.enabled }}
     export EXTERNAL_ACCESS_HOST="${MY_POD_IP}"
+    {{- else if .Values.externalAccess.service.domain }}
+    export EXTERNAL_ACCESS_HOST={{ .Values.externalAccess.service.domain }}
```

Comment:
Shouldn't this be the first one to be evaluated? 

```
    {{- if .Values.externalAccess.service.domain }}
    export EXTERNAL_ACCESS_HOST={{ .Values.externalAccess.service.domain }}
    {{- else if and .Values.externalAccess.service.usePodIPs .Values.externalAccess.autoDiscovery.enabled }}
    export EXTERNAL_ACCESS_HOST="${MY_POD_IP}"
    ...
```

This way it will use one of the following in this order:
- The domain
- The pod IP
- The host IP
- Or use the curl information

Reply:
Generally speaking, the priority for external access is given to the domain configuration, followed by the host IP configuration. Therefore, I have modified the order accordingly and fixed the bug where the domain setting was not taking effect when automatic discovery was configured

Reply:
https://github.com/bitnami/charts/tree/main/bitnami/kafka#using-nodeport-services
here is the infor:
```
Option A) Use random node ports using an initContainer that discover them automatically.

externalAccess.enabled=true
externalAccess.service.type=NodePort
externalAccess.autoDiscovery.enabled=true
serviceAccount.create=true
rbac.create=true
```